foundations/src/telemetry/server/mod.rs

```diff
@@ -14,18 +14,33 @@ use std::net::SocketAddr;
 use std::pin::Pin;
 use std::sync::Arc;
 use std::task::{Context, Poll};
+use tokio::io::{AsyncRead, AsyncWrite};
 use tokio::net::TcpListener;
+#[cfg(unix)]
+use tokio::net::{TcpStream, UnixListener, UnixStream};
 use tokio::sync::watch;
 
 mod router;
 
 use router::Router;
+
+trait AsyncReadWrite: AsyncRead + AsyncWrite + Send + Unpin + 'static {}
```

Comment:
Why is this needed?

Reply:
I added this to be able to store each kind of listener in the same variable - https://github.com/cloudflare/foundations/pull/128/files#diff-caee81f82e8cb546f9fde63dc42376101dc4498dfcdf5b7af0ddd8890312338fR135-R137